### PR TITLE
ip-domain issue fixed

### LINKED ISSUE #984: Allow installation on both ip and domain by default with installation limit 2

### PR DESCRIPTION
**Pull request by ashu555**
Fixes #984

### CODE REVIEW

File: app/Http/Controllers/License/LicenseController.php [hunk 1]
```diff
@@ -205,17 +204,23 @@ public function updateLicensedDomain($licenseCode, $domain, $productId, $license
...
     protected function getIpAndDomain($domain)
     {
+
         $isIP = (bool) ip2long($domain);
-        if ($isIP == true) {
-            $requiredomain = 0;
-            $ip = $domain;
-            $domain = '';
+        if ($domain != '') {
+            if ($isIP == true) {
```
Comment (avinash403) [inline app/Http/Controllers/License/LicenseController.php:210]: == true isn't needed

Comment (avinash403) [reply app/Http/Controllers/License/LicenseController.php:210]: please follow camelCase convention for naming php variables. (requiredomain => requireDomain)

File: app/Http/Controllers/License/LicenseController.php [hunk 2]
```diff
@@ -205,17 +204,23 @@ public function updateLicensedDomain($licenseCode, $domain, $productId, $license
...
+                $requiredomain = 0;
+                $ip = $domain;
+                $domain = '';
+            } else {
+                $requiredomain = 1;
+                $domain = $domain;
```
Comment (avinash403) [inline app/Http/Controllers/License/LicenseController.php:216]: what is the purpose of this line?

File: app/Http/Controllers/License/LicenseController.php [hunk 3]
```diff
@@ -205,17 +204,23 @@ public function updateLicensedDomain($licenseCode, $domain, $productId, $license
      */
     protected function getIpAndDomain($domain)
```
Comment (avinash403) [inline app/Http/Controllers/License/LicenseController.php:205]: This method can be made more readable. My suggestion:
```
        if($domain != ""){
            if(ip2long($domain)){
                return ["ip"=> $domain, 'domain'=> "", 'requiredomain'=>0]
            } else {
                return ["ip"=> "", 'domain'=> $domain, 'requiredomain'=>1]
            }
        }

        return ['ip'=> "", 'domain'=> $domain, 'requiredomain'=> 0];
```
Add necessary comments to it

File: app/Http/Controllers/License/LicenseController.php (hunk 2 above)
Comment (ashu555) [reply app/Http/Controllers/License/LicenseController.php:216]: If the url is in domain form then 'Allow Require Domain' option in License manager will be 1.

**Comment by ashu555** [comment]
> I checked IP and domain installation on local and my test server working fine. But Please remove Domain option installation preference. No use case.
> https://docs.google.com/document/d/1McsBPEcu2ataQ69AgQC8awWQ1eDXuVK0pzKz9dlsPV4/edit

fixed.

**Pull request merged**